src/navigation/common.py

```diff
@@ -0,0 +1,80 @@
+from abc import ABC
+from typing import Tuple
+
+import numpy as np
+
+import rospy
+import smach
+import tf2_ros
+from geometry_msgs.msg import Twist
+from visualization_msgs.msg import Marker
+
+
+# TODO: rename?
+class Context:
```

Comment:
I think this Context object makes sense, but I also think one more layer of abstraction might be good. E.g. a `Rover` class that stores the current state estimate and updates itself with each tick of the state machine or even just when the user asks for the TF.

Reply:
Agreed